src/components/UserInfo/UserInfo.tsx

```diff
@@ -1 +1,14 @@
-export const UserInfo = () => {};
+import React from 'react';
+import { User } from '../TSTypes/user';
+
+type Props = {
+  user?: User
+};
```

Comment:
```suggestion
type Props = {
  user: User
};
```
You render `UserInfo` component only if you have a user, remove unnecessary checks in this component